module/VuFind/src/VuFind/View/Helper/Root/Record.php

```diff
@@ -857,4 +858,36 @@ protected function deduplicateLinks($links)
     {
         return array_values(array_unique($links, SORT_REGULAR));
     }
+
+    /**
+     * Get the source identifier + unique id of the record without spaces
+     *
+     * @param string $idPrefix Prefix for HTML ids
+     *
+     * @return string
+     */
+    public function getUniqueHtmlElementId($idPrefix = '')
+    {
+        return preg_replace(
+            "/\s+/",
+            '_',
+            ($idPrefix ? $idPrefix . '-' : '')
+            . $this->driver->getResultSetIdentifier() . '-'
```

Comment:
We should add handling here to ensure this value is non-null, since concatenating null into a string may cause a notice. It would probably be good to add a case to the unit test to cover the "null record set identifier" case so we can ensure it behaves as expected.